packages/core-data/src/hooks/use-permissions.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * WordPress dependencies
+ */
+import { useQuerySelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { IDLE, SUCCESS, RESOLVING } from './constants';
+import { store as coreStore } from '../';
+
+export default function usePermissions( resource, id ) {
```

Comment:
Nit: for consistency, it might use a more explicit name like `useEntityRecordPermissions` since it works only for a single record. 

What's the story for multiple records?